src/common/services/api/designations.service.spec.js

```diff
@@ -90,58 +92,63 @@ describe('designation service', () => {
       frequency: 'NA',
       displayName: 'Steve Peck',
       designationType: 'Staff',
+      orgId: 'STAFF',
       code: '0354433',
       designationNumber: '0354433'
     }
-    it('should get product details for a designation number', () => {
+    it('should get product details for a designation number', done => {
       self.$httpBackend.expectPOST('https://give-stage2.cru.org/cortex/items/crugive/lookups/form?FollowLocation=true&zoom=code,offer:code,definition,definition:options:element:selector:choice,definition:options:element:selector:choice:description,definition:options:element:selector:choice:selectaction,definition:options:element:selector:chosen,definition:options:element:selector:chosen:description',
         { code: '0354433' })
         .respond(200, lookupResponse)
       self.designationsService.productLookup('0354433')
         .subscribe((data) => {
           expect(data).toEqual(expectedResponse)
-        })
+          done()
+        }, done)
```

Comment:
I'm actually intentionally calling `done` with the error if the observable errors. That will immediately fail the test and print the error instead of the test eventually timing out without printing any error information.

That pattern is described here: https://jestjs.io/docs/asynchronous#callbacks

Reply:
Then we should call `fail` instead of `done` because otherwise it will succeed and give us false test successes.

Reply:
Apparently, `fail` isn't defined in jest tests. It seems that that is a jasmine thing. Calling `fail()` in a test will technically fail the test, but only because it throws:

```
ReferenceError: fail is not defined
```

I probably wasn't explaining it well before but calling `done()` will complete the test successfully, but calling `done(err)` or `done("message")` will fail the test immediately. When I do `observable.subscribe(() => {}, done)`, that is equivalent to `observable.subscribe(() => {}, err => done(err))`. If the observable errors, `done` will be called with an argument, and the test will immediately fail, which is what we want.

Reply:
Ah right, forgot about that `done(message)` implementation. I suppose I'm still more used to Jasmine than Jest.